recipes/date/all/test_package/conanfile.py

```diff
@@ -8,10 +8,11 @@ class TestPackageConan(ConanFile):
 
     def build(self):
         cmake = CMake(self)
+        cmake.definitions["DATE_HEADER_ONLY"] = self.options["date:header_only"]
```

Comment:
Add this to `cpp_info.defines` in the recipe. Other consumers will need to define this too.

Reply:
No they won't need it. I just invented this define in order to disable tests of some features that doesn't work in header_only mode.

Reply:
I've checked DATE_HEADER_ONLY, curl and pthread. They are not listed on headers.